forensics/Copy-VmDigitalEvidence.ps1

```diff
@@ -42,152 +42,190 @@ $ErrorActionPreference = 'Stop'
-$destSA = 'PLACEHOLDER'                               # The name of the storage account
+$destSAblob = 'PLACEHOLDER'                           # The name of the storage account for BLOB
+$destSAfile = 'PLACEHOLDER'                           # The name of the storage account for FILE
 $destTempShare = 'PLACEHOLDER'                        # The temporary file share mounted on the hybrid worker
 $destSAContainer = 'PLACEHOLDER'                      # The name of the container within the storage account
 $destKV = 'PLACEHOLDER'                               # The name of the keyvault to store a copy of the BEK in the dest subscription
 
-$targetLinuxDir = "/mnt/$destSA/$destTempShare"       # The name of directory in which file share is mounted
-$snapshotPrefix = (Get-Date).toString('yyyyMMddhhmm') # The prefix of the snapshot to be created
+$targetLinuxDir = "/mnt/$destSAfile/$destTempShare"       # The name of directory in which file share is mounted 
+$snapshotPrefix = (Get-Date).toString('yyyyMMddHHmm') # The prefix of the snapshot to be created
 
 #############################################################################################
-
-################################## Login session ############################################
-# Connect to Azure (via Managed Identity or Azure Automation's RunAs Account)
-#
-# Feel free to adjust the following lines to invoke Connect-AzAccount via
-# whatever mechanism your Hybrid Runbook Workers are configured to use.
-#
-# Whatever service principal is used, it must have the following permissions
-#  - "Contributor" on the Resource Group of target Virtual Machine. This provide snapshot rights on Virtual Machine disks
-#  - "Storage Account Contributor" on the immutable SOC Storage Account
-#  - Access policy to Get Secret (for BEK key) and Get Key (for KEK key, if present) on the Key Vault used by target Virtual Machine
-#  - Access policy to Set Secret (for BEK key) and Create Key (for KEK key, if present) on the SOC Key Vault
-
-Add-AzAccount -Identity
-
-############################# Snapshot the OS disk of target VM ##############################
+################################## Hybrid Worker Check ######################################
 Write-Output "#################################"
 Write-Output "Snapshot the OS Disk of target VM"
 Write-Output "#################################"
+$bios=Invoke-Expression -Command "dmidecode -t bios -q"
+if ($bios) {   
+    Write-Output "Running on Hybrid Worker"
+
+    ################################## Mounting fileshare #######################################
+    # The Storage account also hosts an Azure file share to use as a temporary repository for calculating the snapshot's SHA-256 hash value.
+    # The following doc shows a possible way to create a persistent mount point:
+    # https://docs.microsoft.com/en-us/azure/storage/files/storage-how-to-use-files-linux#mount-the-azure-file-share-on-demand-with-mount

```

Comment:
makes sure we're handling user locale preferences by removing the locale part of the URL.

```suggestion
    # https://docs.microsoft.com/azure/storage/files/storage-how-to-use-files-linux#mount-the-azure-file-share-on-demand-with-mount
```

Reply:
ok, language "en-us" removed from the link

Reply:
👍